### PR TITLE
Int v0.8 - updated conditional logic to prevent hover e interactions on proximity

### PR DESCRIPTION
**Pull request by MbfloydIR**
## Summary
https://tsu.atlassian.net/browse/IR-4320

## References
closes #_insert number here_

### CODE REVIEW

**Review by SamMazerIR — APPROVED** [review]

File: packages/engine/src/interaction/components/InteractableComponent.ts [hunk 1]
```diff
@@ -323,7 +324,13 @@ export const InteractableComponent = defineComponent({
     InputComponent.useExecuteWithInput(
       () => {
         const buttons = InputComponent.getMergedButtons(entity)
-        if (!interactableComponent.clickInteract.value && buttons.PrimaryClick?.pressed) return
+        if (
+          !activateUI ||
```
Comment (HexaField) [inline packages/engine/src/interaction/components/InteractableComponent.ts]: This is scoping a global variable from a different context. We need a different solution.

**Review by HexaField — DISMISSED** [review]

File: packages/engine/src/interaction/components/InteractableComponent.ts (hunk 1 above)
Comment (MbfloydIR) [reply packages/engine/src/interaction/components/InteractableComponent.ts]: implemented

File: packages/engine/src/interaction/components/InteractableComponent.ts [hunk 2]
```diff
@@ -272,6 +273,8 @@ export const InteractableComponent = defineComponent({
     if (json.label) component.label.set(json.label)
     if (typeof json.uiActivationType === 'number' && component.uiActivationType.value !== json.uiActivationType)
       component.uiActivationType.set(json.uiActivationType)
+    if (typeof json.canInteract === 'boolean' && component.canInteract.value !== json.canInteract)
+      component.canInteract.set(json.canInteract)
```
Comment (SamMazerIR) [inline packages/engine/src/interaction/components/InteractableComponent.ts]: I don't think we need to serialize canInteract. It should always represent a current state.

**Review by SamMazerIR — CHANGES_REQUESTED** [review]
lets remove serialization of canInteract

File: packages/engine/src/interaction/components/InteractableComponent.ts (hunk 2 above)
Comment (MbfloydIR) [reply packages/engine/src/interaction/components/InteractableComponent.ts]: we need to serialize it so the user can check/uncheck the can interact editor ui checkbox, as well as loading the saved value when the scene is loaded

**Review by SamMazerIR — APPROVED** [review]

**Pull request merged**